stl/src/print.cpp

```diff
@@ -0,0 +1,280 @@
+    // return value of GetConsoleMode().
+    DWORD _Console_mode;
+    const bool _Is_unicode_console = GetConsoleMode(_Console_handle, &_Console_mode) != 0;
+
+    if (!_Is_unicode_console) {
+        return __std_unicode_console_retrieval_result{._Error = __std_win_error::_File_not_found};
+    }
+
+    return __std_unicode_console_retrieval_result{
+        ._Console_handle = static_cast<__std_unicode_console_handle>(
+            reinterpret_cast<_STD underlying_type_t<__std_unicode_console_handle>>(_Console_handle)),
+        ._Error = __std_win_error::_Success};
+}
+
+_END_EXTERN_C
+
+namespace {
+    class _Allocated_string {
+    public:
+        _Allocated_string() = default;
+
+        explicit _Allocated_string(__crt_unique_heap_ptr<wchar_t>&& _Other_str, const size_t _Other_capacity) noexcept
+            : _Str(_STD move(_Other_str)), _Str_capacity(_Other_capacity) {}
+
+        [[nodiscard]] wchar_t* _Data() const noexcept {
+            return _Str.get();
+        }
+
+        [[nodiscard]] size_t _Capacity() const noexcept {
+            return _Str_capacity;
+        }
+
+        void _Reset() noexcept {
+            _Str.release();
+            _Str_capacity = 0;
+        }
+
+    private:
+        __crt_unique_heap_ptr<wchar_t> _Str;
+        size_t _Str_capacity = 0;
+    };
+
+    template <class _Char_type>
+    class _Really_basic_string_view {
+    public:
+        _Really_basic_string_view() = default;
+
+        explicit _Really_basic_string_view(const _Char_type* const _Other_str, const size_t _Other_size) noexcept
+            : _Str(_Other_str), _Str_size(_Other_size) {}
+
+        [[nodiscard]] const _Char_type* _Data() const noexcept {
+            return _Str;
+        }
+
+        [[nodiscard]] size_t _Size() const noexcept {
+            return _Str_size;
+        }
+
+        [[nodiscard]] bool _Empty() const noexcept {
+            return _Str_size == 0;
+        }
+
+    private:
+        const _Char_type* _Str = nullptr;
+        size_t _Str_size       = 0;
+    };
+
+    using _Minimal_string_view  = _Really_basic_string_view<char>;
+    using _Minimal_wstring_view = _Really_basic_string_view<wchar_t>;
+
+    constexpr size_t _Max_str_segment_size = 8192;
+
+    [[nodiscard]] _Minimal_string_view _Get_next_utf8_string_segment(
+        const char* const _Str, const size_t _Str_size) noexcept {
+        if (_Str_size <= _Max_str_segment_size) [[likely]] {
+            return _Minimal_string_view{_Str, _Str_size};
+        }
+
+        // We want to find a pointer to the last valid code point _End_ptr such that the number of
+        // bytes in [_Str, _End_ptr] is <= _Max_str_segment_size. However, we also want to make sure
+        // that we end on a complete grapheme cluster to ensure that WriteConsoleW() can correctly
+        // draw Unicode characters which consume more than one code point.
+        _STD _Grapheme_break_property_iterator<char> _Grapheme_cluster_itr{_Str, _Str + _Str_size};

```

Comment:
> This is included from a header which includes `<xutility>`, so we can't use this.

This is just because of the use of `upper_bound` and `_STL_INTERNAL_CHECK` (and `default_sentinel`?).
I guess it'd be trivial enough to use a simplified, unchecked variant of `upper_bound`, and remove all `_STL_INTERNAL_CHECK` in `<__msvc_format_ucd_tables.hpp>` and `<__msvc_unicode_iterators.hpp>` to make the headers core.